backend/src/Service/Admin/Order/AdminOrderService.php

```diff
@@ -1954,4 +1960,127 @@ public function getCaptainsWhoDeliveredLastFiveOrders(): array
+            $orderCreateByAdminRequest->setIsHide(OrderIsHideConstant::ORDER_SHOW);
+
+            if ($canCreateOrder->subscriptionStatus === SubscriptionConstant::CARS_FINISHED) {
+                $orderCreateByAdminRequest->setIsHide(OrderIsHideConstant::ORDER_HIDE_TEMPORARILY);
+            }
+
+            // 5 set the delivery date equals to current datetime + 3 minutes just for affording the late in persisting
+            // the order to the database if it is happened
+            $orderCreateByAdminRequest->setDeliveryDate((new DateTime('+ 3 minutes'))->format('Y-m-d H:i:s'));
+
+            $orderCreateByAdminRequest->setPayment(OrderTypeConstant::ORDER_PAYMENT_CASH);
+            $orderCreateByAdminRequest->setOrderCost(OrderCostDefaultValueConstant::ORDER_VALUE_CONST);
+            $orderCreateByAdminRequest->setNote(OrderNoteDefaultValueConstant::ORDER_NOTE_DEFAULT_VALUE_CONST);
+            $orderCreateByAdminRequest->setDestination(OrderClientLocationConstant::ORDER_CLIENT_LOCATION_ARRAY_CONST[array_rand(OrderClientLocationConstant::ORDER_CLIENT_LOCATION_ARRAY_CONST)]);
+            $orderCreateByAdminRequest->setRecipientName(OrderRecipientDefaultValueConstant::ORDER_RECIPIENT_NAME_DEFAULT_VALUE_CONST);
+            $orderCreateByAdminRequest->setRecipientPhone(OrderRecipientDefaultValueConstant::ORDER_RECIPIENT_PHONE_DEFAULT_VALUE_CONST);
+            $orderCreateByAdminRequest->setDetail(OrderNoteDefaultValueConstant::ORDER_NOTE_DEFAULT_VALUE_CONST);
+
+            // 6 calculate the distance between the store's branch and the client location
+            $distance = $this->getDistanceBetweenTwoLocations($branch->getLocation(), $orderCreateByAdminRequest->getDestination());
+            $orderCreateByAdminRequest->setStoreBranchToClientDistance($distance);
+
+            // 7 calculate the delivery cost
+            $orderCreateByAdminRequest->setDeliveryCost($this->calculateDeliveryCostByStoreOwnerProfileIdAndDistance($storeOwnerProfile->getId(),
+                $distance));
+
+            // 8 create order record
+            $order = $this->adminOrderManager->createOrderByAdmin($orderCreateByAdminRequest);
+
+            if ($order) {
+                // 9 update the remaining orders of the store's subscription
+                $this->updateRemainingOrdersOfStoreSubscriptionByAdmin($storeOwnerProfile->getId(), $order->getCreatedAt(),
+                    SubscriptionConstant::OPERATION_TYPE_SUBTRACTION, 1);
+
+                // 10 create local notification for the store
+                $this->createLocalNotificationForStore($storeOwnerProfile->getId(), NotificationConstant::NEW_ORDER_TITLE,
+                    NotificationConstant::CREATE_ORDER_SUCCESS, $order->getId());
+
+                // 11 create order timeline
+                $this->createOrderLogViaOrderEntity($order);
+
+                // 12 save log of the action on order
+                $this->createOrderLogMessageViaOrderEntityAndByAdmin($order, $userId);
```

Comment:
What is this?

Reply:
For creating order log via Symfony Messenger.